index.js

```diff
@@ -41,10 +41,10 @@ module.exports = {
     if (process.env.EMBER_CLI_FASTBOOT) {
       process.env.EMBER_CLI_FASTBOOT_APP_NAME = app.name;
       app.options.autoRun = false;
+      
+      // We serve the index.html from fastboot-dist, so this has to apply to both builds
+      app.options.storeConfigInMeta = false;
```

Comment:
We solve this by having config/environment branch, on the server side looking for it on the sandbox and on the client side looking for it in meta.  I agree with @stefanpenner we should not be  building app.js twice once for the server and once for the client.